src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -661,22 +662,22 @@ internal static T[] AllocateUninitializedArray<T>(int length)
             }
 
             if (length < 0)
```

Comment:
This should be in the unmanaged FCall - no need to force-inline the error handling everywhere. Also, we should make the exception same as the exception thrown by `new T[length]` for negative length.

Reply:
FCall passes everything to the same helper as for normal arrays (just with a noinit flag). 
It all was refactored to go through `AllocateSzArray` internally, which has the checks for sizes, dimensions etc...

The negative length check is not necessary here. I do not remember why it was added. Do we care from where the exception would be thrown? (it think not)

Reply:
Right, we do not care - as long as the exception is consistent.